src/node.rs

```diff
@@ -41,47 +41,47 @@ use {
 };
 
 /// A builder to configure and create a new `Node`.
-pub struct NodeBuilder {
+pub struct Builder {
     first: bool,
     rng: Option<MainRng>,
     network_config: Option<NetworkConfig>,
     full_id: Option<FullId>,
     network_cfg: NetworkParams,
 }
 
-impl NodeBuilder {
+impl Builder {
     /// Configures the node to start a new network instead of joining an existing one.
     pub fn first(self, first: bool) -> Self {
-        Self { first, ..self }
+        Builder { first, ..self }
```

Comment:
Why not use `Self` to avoid repeating the name? There is also a clippy lint for that which I think is enabled?

Reply:
> There is also a clippy lint

It's [use_self](https://rust-lang.github.io/rust-clippy/master/) but it's not enabled because it has some annoying false positives right now. I'd still prefer to use `Self` instead of repeating the type name, but as we don't enforce it via a lint, I don't insist on it either.

Reply:
it seem some of the previous PRs converted TypeName to Self, so that seem to be a move backward, any reason for it?

Reply:
Yes, Self is best here. I will change it. I don't think this PR reverted that though it just never changed it to "fix" the repeated typename.